src/clj/puppetlabs/services/request_handler/request_handler_service.clj

```diff
@@ -3,18 +3,28 @@
             [puppetlabs.services.protocols.request-handler :as handler]
             [puppetlabs.services.request-handler.request-handler-core :as core]
             [puppetlabs.services.jruby.jruby-puppet-service :as jruby]
-            [puppetlabs.trapperkeeper.services :as tk-services]))
+            [puppetlabs.trapperkeeper.services :as tk-services]
+            [clojure.tools.logging :as log]))
 
 (defn- handle-request
-  [request jruby-service config]
-  (jruby/with-jruby-puppet jruby-puppet jruby-service
-    (core/handle-request request jruby-puppet config)))
+  [request jruby-service config puppet-version]
+  (try
+    (jruby/with-jruby-puppet jruby-puppet jruby-service
+                             (core/handle-request request jruby-puppet config))
+    (catch IllegalStateException e
+      (let [message (.getMessage e)]
+        (log/error message)
+        {:status  503
+         :body    message
+         :headers {"Content-Type"     "text/plain"
+                   "X-Puppet-version" puppet-version}}))))
```

Comment:
Might be good to take the `wrap-with-puppet-version-header` middleware in the `certificate-authority-core` namespace and put it into a common place - maybe clj.puppetlabs.pupetserver.ringutils? - that both the CA and this function could call into to get the same wrapping?